CRT constraints and AMM benchmarks

dobertRowneySr opened the pull request:
fixes:
- max yearly patronage rate 
- patronage rate computation & approximation updated
- patronage tests refactored & updated
- add max limit for transfers
- adds benchmarks for amm (as #4481 has lost all changes)

## Resources and References:
- [summary table with CRT governance constraints](https://github.com/Joystream/joystream/issues/4151#issuecomment-1455821245) 
- [AMM manual page reference](https://handbook.joystream.org/system/content-directory/projecttoken#amm)
- [AMM easy to read summary (Bonding Curve means our AMM functionality)](https://medium.com/linum-labs/intro-to-bonding-curves-and-shapes-bf326bc4e11a)
- [Update patronage rate computation](https://github.com/Joystream/joystream/issues/4378)

dobertRowneySr committed:
feat: add bm for uby on amm

dobertRowneySr committed:
feat: add benchmark for `sell_on_amm`

dobertRowneySr committed:
chore: add `activate_amm` bm

dobertRowneySr committed:
feat: add `deactivate_amm` bm

dobertRowneySr committed:
feat: add tests on benchmarks

dobertRowneySr committed:
feat: add weights on project token

dobertRowneySr committed:
fix: cargo test errors on benchmarks

dobertRowneySr committed:
fix: benchmarks
I used a big endian representation for the membership handle as
the previous technique used to throw overflow error

dobertRowneySr committed:
feat: generate weights

dobertRowneySr committed:
feat: add max yearly patronage rate

dobertRowneySr committed:
feat: add propsal for patronage rate update

dobertRowneySr committed:
fix: bound input on multi-transfer

dobertRowneySr committed:
fix: deadline & max tranfer bound

dobertRowneySr committed:
fix: transfer outputs now statically bounded

dobertRowneySr committed:
fix: patronage rate computation

dobertRowneySr committed:
refactor: patronage test

dobertRowneySr committed:
fix: interest rate approximation

dobertRowneySr committed:
fix: all tests passing

dobertRowneySr committed:
fix: patronage rate computation

dobertRowneySr committed:
chore: add test with >100% rate

dobertRowneySr committed:
chore: add some explanatory comments

Commit:
feat: add test for slope parameter too low

Commit:
feat: add slope parameter check & fix test values

Commit:
fix: adjust content pallet & fix warnings

Commit:
fix: adjust chain_spec file with slope parameter

Commit:
fix: build error

Commit:
fix: add numerical computation

Commit:
fix: integrate computation with token pallet

Commit:
fix: failing tests & build errors

Commit:
fix: failing tests & build errors

Commit:
fix: failing tests

Commit:
refactor: type traits

Commit:
fix: build errors

Commit:
fix: benchmarks cargo checks errors

Commit:
fix: build error and generte types and metadata

Commit:
fix: linter

mnaamani committed:
bump joystream-node version to v8.3.1

mnaamani committed:
runtime upgrade test workflow only if changes to runtime/src/lib.rs and runtime/Cargo.toml

Commit:
fix: misc error after merged

Commit:
fix: linter

dobertRowneySr commented on runtime-modules/project-token/src/lib.rs:
```diff
@@ -819,16 +825,15 @@ decl_module! {
         /// - event deposited
-        #[weight = 100_000_000] // TODO: adjust weight
-        fn buy_on_amm(origin, token_id: T::TokenId, member_id: T::MemberId, amount: <T as Config>::Balance, deadline: Option<<T as timestamp::Config>::Moment>, slippage_tolerance: Option<(Permill, JoyBalanceOf<T>)>) -> DispatchResult {
+        #[weight = WeightInfoToken::<T>::buy_on_amm_with_existing_account()]
```
Replace this with a `max(buy_on_amm_with_existing_account, buy_on_amm_with_new_account)`
as the weight might differ on the ref. machine

dobertRowneySr commented on runtime-modules/project-token/src/lib.rs:
```diff
@@ -972,6 +966,19 @@ decl_module! {
             Ok(())
         }
 
+        #[weight = 100_000_000] // TODO: adjust weight
```
might as well benchmark this

dobertRowneySr commented on runtime-modules/project-token/src/lib.rs:
```diff
@@ -913,10 +912,10 @@ decl_module! {
         /// - respective JOY amount transferred from amm treasury account to user account
         /// - event deposited
-        #[weight = 100_000_000] // TODO: adjust weight
-        fn sell_on_amm(origin, token_id: T::TokenId, member_id: T::MemberId, amount: <T as Config>::Balance, deadline: Option<<T as timestamp::Config>::Moment>, slippage_tolerance: Option<(Permill, JoyBalanceOf<T>)>) -> DispatchResult {
```
This is good, timestamp info can be manipulated by validator node (https://neptunemutual.com/blog/understanding-block-timestamp-manipulation/)

mnaamani replied:
It doesn't look like the deadline was used for any purpose anyway, so it makes sense to drop it.

dobertRowneySr commented on runtime-modules/project-token/src/lib.rs:
```diff
@@ -247,20 +245,28 @@ decl_module! {
             metadata: Vec<u8>
         ) -> DispatchResult {
+
+            // // security check
```
Remove this if dead code

dobertRowneySr replied:
Code is no longer neeed due to Bounded Vec type use as parameter

dobertRowneySr commented on runtime-modules/project-token/src/tests/patronage.rs:
```diff
@@ -264,271 +173,245 @@ fn decreasing_patronage_rate_fails_with_target_rate_exceeding_current_rate() {
     })
 }
 
+// TODO(mrbovo): allow for more than 100% claim over supply
```
no longer relevant: remove this

dobertRowneySr replied:
```suggestion
```

dobertRowneySr commented on runtime-modules/project-token/src/tests/patronage.rs:
```diff
@@ -264,271 +173,245 @@ fn decreasing_patronage_rate_fails_with_target_rate_exceeding_current_rate() {
 }
 
 #[test]
 fn claim_patronage_ok_with_credit_accounting() {
```
```suggestion
fn claim_patronage_ok_with_correct_patronage_credit_accounting() {
```

dobertRowneySr commented on runtime/src/lib.rs:
```diff
@@ -886,6 +886,7 @@ parameter_types! {
     pub const ProjectTokenModuleId: PalletId = PalletId(*b"mo:token"); // module: token
     pub const MaxVestingSchedulesPerAccountPerToken: u32 = 5;
     pub const BlocksPerYear: u32 = 5259600; // 365,25 * 24 * 60 * 60 / 6
+    pub const MaxOutputs: u32 = 256; // TODO(Martin, Ignazio) : find a suitable value
```
Need a separate issue for:
- max patronage rate
- min amm slope

dobertRowneySr commented on runtime-modules/project-token/src/types.rs:
```diff
@@ -871,6 +892,11 @@ impl<Balance, VestingScheduleParams>
 #[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, TypeInfo)]
 pub struct Transfers<MemberId, Payment>(pub BTreeMap<MemberId, Payment>);
 
+// #[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, TypeInfo)]
```
Test code, remove this

dobertRowneySr reviewed (commented):
Some fixes are needed

Commit:
Update runtime-modules/project-token/src/tests/patronage.rs

Commit:
Update runtime-modules/project-token/src/tests/patronage.rs

dobertRowneySr commented on bin/node/src/chain_spec/project_token_config.rs:
```diff
@@ -11,6 +11,8 @@ pub fn production_config() -> ProjectTokenConfig {
         min_revenue_split_duration: days!(21),
         min_revenue_split_time_to_start: 0,
         sale_platform_fee: Permill::from_percent(2),
+        max_yearly_patronage_rate: Permill::from_percent(5).into(),
```
no migration is needed for these parameters as they are new with V2002

dobertRowneySr commented on runtime-modules/content/src/benchmarks/benchmarking.rs:
```diff
@@ -1776,6 +1776,70 @@ benchmarks! {
             );
         }
 
+    activate_amm {
```
Consider adding a comment about why member channel all max doesn't make op. more expensive

dobertRowneySr replied:
curator channel operations seems to be more expensive due to permission checking

dobertRowneySr commented on runtime-modules/content/src/lib.rs:
Ok it appears no (data) migrations are needed here on content pallet

dobertRowneySr commented on runtime-modules/project-token/src/tests/patronage.rs:
```diff
@@ -264,271 +173,244 @@ fn decreasing_patronage_rate_fails_with_target_rate_exceeding_current_rate() {
-    let config = GenesisConfigBuilder::new_empty()
-        .with_token_and_owner(token_id, params.build(), owner_id, supply)
-        .build();
+fn claim_patronage_ok_with_correct_credit_accounting_and_more_than_100_percent_supply() {
```
This also accounts for testing numeric precision, however there are several tests in the `common/numerical.rs` library for this

dobertRowneySr commented on query-node/mappings/src/proposals.ts:
```diff
@@ -333,9 +337,15 @@ async function handleRuntimeUpgradeProposalExecution(event: SubstrateEvent, stor
+  const specVersion = block.runtimeVersion.specVersion
   const [proposalId, generalProposalParameters, runtimeProposalDetails, proposalThreadId] =
-    new ProposalCreatedEvent_V1001(event).params
+    specVersion >= 2002 ? new ProposalCreatedEvent_V2002(event).params : new ProposalCreatedEvent_V1001(event).params
```
Should we update the mappings in order to account for the new update max yearly patronage rate proposal?
If so I need to open an issue about this

mnaamani replied:
Will it depends on when we decide to enable creating tokens, or start adding integration tests then yes we would need to update the mappings starting with handling new proposal type in `parseProposalDetails()`

mnaamani committed:
github workflow - fix use of 'git-diff-action'

mnaamani committed:
runtime upgrade workflow - only test .rs file for changes

Commit:
fix: crt benchmarks

Commit:
fix: content benchmarks

Commit:
fix: remove dead code

Commit:
fix: macro

mnaamani commented on bin/node/src/chain_spec/project_token_config.rs:
```diff
@@ -11,6 +11,9 @@ pub fn production_config() -> ProjectTokenConfig {
         min_revenue_split_time_to_start: 0,
         sale_platform_fee: Permill::from_percent(2),
+        max_yearly_patronage_rate: Permill::from_percent(20).into(), // TODO: update, raising might make benchmarks fail
+        min_amm_slope_parameter: 10, // TODO: update, raising might make benchmarks fail
```
The default storage value set in project-token decl_storage macro is 15% for max_yearly_patronage_rate and 0 for min_amm_slope_parameter. In a runtime upgrade those defaults will be used because they are now being introduced in nara upgrade. If we are not happy with the defaults, we need to add a migration on_runtime_upgrade.

mnaamani commented on runtime-modules/common/src/numerical.rs:
```diff
@@ -0,0 +1,156 @@
+    FixedPointNumber, FixedU128, PerThing, Permill, Perquintill,
+};
+
+pub const ORDER: usize = 20;
```
could you comment what this  constant stands for?

Commit:
fix: error

Commit:
fix: TryInto error

Commit:
Fix: Cache cleanup not working for objects in group `0`

Commit:
Merge pull request #4752 from kdembler/elastic-logs-auth

[Argus] add auth options to elastic logs config

Commit:
fix: timestamp not needed

Commit:
fix: macro

Commit:
fix: macro days

Commit:
fix: macro

mnaamani committed:
Merge pull request #4755 from mnaamani/update-librocksdb-sys

Update librocksdb-sys from v0.6.1+6.28.2 to v0.6.3+6.28.2

Commit:
fix: macro

kdembler committed:
add workerId to argus status response

kdembler committed:
[Argus] allow specifying elastic logs index

Commit:
fix: clippy

mnaamani commented on runtime/src/proposals_configuration/testing.rs:
```diff
@@ -331,3 +331,18 @@ pub(crate) fn update_channel_payouts_proposal() -> ProposalParameters<BlockNumbe
+pub(crate) fn update_max_yearly_patronage_rate_proposal() -> ProposalParameters<BlockNumber, Balance>
+{
+    ProposalParameters {
+        voting_period: days!(7),
```
voting and grace periods for testing,playground need to be much lower. See values for other proposals.

mnaamani commented on runtime/src/weights/block_weights.rs:
```diff
@@ -16,23 +16,22 @@
 
 //! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 4.0.0-dev
-//! DATE: 2022-10-20 (Y/M/D)
+//! DATE: 2023-05-11 (Y/M/D)
```
block, extrinsics and rocksdb _weights.rs should not be modified in this PR.

mnaamani commented on runtime-modules/content/src/weights.rs:
```diff
@@ -807,20 +801,42 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
+	// Storage: Token TokenInfoById (r:1 w:1)
+	// Storage: Token MinAmmSlopeParameter (r:1 w:0)
+	// Storage: System Account (r:1 w:1)
+	fn activate_amm() -> Weight {
```
perhaps we can keep only the new weight functions (and maybe issue_new_token which was modified) in this file and checkout all other changes.
We will do a final weight generation step before nara release.

mnaamani commented on runtime-modules/content/src/benchmarks/mod.rs:
```diff
@@ -79,15 +79,15 @@ const MAX_KILOBYTES_METADATA: u32 = 100;
 // Creator tokens
 const MAX_CRT_INITIAL_ALLOCATION_MEMBERS: u32 = 1024;
-const MAX_CRT_ISSUER_TRANSFER_OUTPUTS: u32 = 1024;
+const MAX_CRT_ISSUER_TRANSFER_OUTPUTS: u32 = 256; // same as in Token MaxOuputs
```
If it is a runtime constant, perhaps it can be read from the pallet instead of re-defining a separate constant and having to remember to keep them in sync?

mnaamani reviewed (commented):
I left just an initial high level review, did not go into implementation details yet.

Commit:
Adjust Orion docker setup: Use separate db service

Commit:
fix: testing propsals

Commit:
fix: playground proposal config

mnaamani committed:
Merge pull request #4757 from Lezek123/argus-fix-cache-drop

Argus fix: Cache cleanup not working properly for objects in group `0`

mnaamani committed:
Merge pull request #4730 from Lezek123/orion-v2-docker-setup

Orion v2 docker setup

mnaamani commented on runtime-modules/common/src/numerical.rs:
```diff
@@ -0,0 +1,156 @@
+pub const ORDER: usize = 20;
+
+// does not work with 100% interest
+pub fn natural_log_1_plus_x(interest: Permill) -> Perquintill {
```
I see this is not used expect internally from `one_plus_interest_pow_frac` so maybe keep it private?
Similarly for `one_plus_interest_pow_frac`.

If this function "does not work with 100% interest" would would the output be, and should we guard against invalid input?

mnaamani commented on runtime-modules/common/src/numerical.rs:
```diff
@@ -0,0 +1,156 @@
+    result
+}
+
+pub fn one_plus_interest_pow_frac(interest: Permill, exp: Perquintill) -> FixedU128 {
```
Can you add rust-doc just to show what is being calculated

mnaamani commented on runtime-modules/project-token/src/benchmarking.rs:
```diff
@@ -28,15 +30,18 @@ const DEFAULT_TOKENS_ON_SALE: u32 = 100_000;
+
+// Amm
+const DEFAULT_AMM_AMOUNT: u32 = 1000;
+const DEFAULT_AMM_jOY_AMOUNT: u32 = 5_100_000; // (a = 10) * amount^2 /2 + (b = 100) * amount, amount = 1000
```
```suggestion
const DEFAULT_AMM_JOY_AMOUNT: u32 = 5_100_000; // (a = 10) * amount^2 /2 + (b = 100) * amount, amount = 1000
```

mnaamani commented on runtime-modules/project-token/src/benchmarking.rs:
```diff
@@ -28,15 +30,18 @@ const DEFAULT_TOKENS_ON_SALE: u32 = 100_000;
-
-// Patronage
-const DEFAULT_PATRONAGE: YearlyRate = YearlyRate(Permill::from_percent(1));
+const DEFAULT_SPLIT_ALLOCATION: u32 = 4_000_000; // DEFAULT_REVENUE_SPLIT_RATE * DEFAULT_SPLIT_REVENUE
```
Why not evaluate it as per comment`DEFAULT_REVENUE_SPLIT_RATE * DEFAULT_SPLIT_REVENUE`

dobertRowneySr replied:
because `DEFAULT_PATRONAGE: Permill` and the multiplaction is a non constant operation, so rustc will throw error

mnaamani commented on runtime-modules/project-token/src/benchmarking.rs:
```diff
@@ -237,26 +265,18 @@ benchmarks! {
-            })
-            .collect()
-        );
+        let mut outputs = TransferOutputsOf::<T>::with_bounded_capacity((o + 1) as usize);
```
by adding + 1 are not going over the `MAX_TX_OUTPUTS`. range of `o` starts at 1, so I don't see the need to add +1 if we are concerned calling `with_bounded_capacity` with args `0`..

mnaamani commented on runtime-modules/content/src/benchmarks/mod.rs:
```diff
@@ -227,19 +228,20 @@ fn get_byte(num: u64, byte_number: u8) -> u8 {
+    // let mut handle = vec![];
 
-    for i in 0..16 {
-        handle.push(get_byte(id, i));
```
this is the only use of `get_byte()`, if we are sure it is not needed anymore, then we can remove commented code and drop implementation of get_byte()

mnaamani commented on runtime-modules/project-token/src/lib.rs:
```diff
@@ -247,20 +245,22 @@ decl_module! {
             )?;
 
             // Currency transfer preconditions
+            // let btree_outputs = Transfers(outputs.into_iter().map(|(member_id, amount)| (member_id, PaymentWithVesting::from(amount))).collect::<BTreeMap<_,_>>());
```
drop unused code?

mnaamani commented on runtime-modules/project-token/src/lib.rs:
```diff
@@ -176,6 +168,12 @@ decl_storage! { generate_storage_info
+        pub MaxYearlyPatronageRate get(fn max_yearly_patronage_rate) config(): YearlyRate = YearlyRate(Permill::from_percent(15));
+
+        /// Minimum slope parameters allowed for AMM curve
+        pub MinAmmSlopeParameter get(fn min_amm_slope_parameter) config(): TokenBalanceOf<T>;
```
there is no need for a disptachable call to set/update this value ?

dobertRowneySr replied:
This will be taken care via proposal (next PR on the roadmap)

mnaamani commented on runtime-modules/project-token/src/tests/amm.rs:
```diff
@@ -63,7 +63,7 @@ fn amm_buy_fails_with_member_and_origin_auth() {
 fn amm_buy_succeeds_with_new_user() {
     let token_id = token!(1);
-    let ((user_member_id, user_account_id), user_balance) = (member!(2), joy!(5_000_000));
+    let ((user_member_id, user_account_id), user_balance) = (member!(2), joy!(5_000_000_000));
```
All tests are updated to give member a larger initial balance. Assuming tests were failing. Can we use a constant that might highlight what and why there is a minimum for tests to pass?

mnaamani commented on runtime-modules/project-token/src/tests/amm.rs:
```diff
@@ -810,10 +786,14 @@ fn deactivate_fails_with_invalid_token_id() {
     // give enough balance to bond amount
-    let ((user_member_id, user_account_id), user_balance) = (member!(2), joy!(10_000_000_000));
+    let ((user_member_id, user_account_id), user_balance) =
+        (member!(2), joy!(10_000_000_000_000_000_000));
```
How is are these values being selected?

mnaamani commented on runtime-modules/project-token/src/tests/amm.rs:
```diff
@@ -810,10 +786,14 @@ fn deactivate_fails_with_invalid_token_id() {
         .execute_with(|| {
-            IssueTokenFixture::default().execute_call().unwrap();
+            IssueTokenFixture::default()
+                .with_supply(1_000u128)
```
how is it determined that 1_000 will result in the expected error?

mnaamani commented on runtime-modules/project-token/src/tests/canonical.rs:
```diff
@@ -1133,37 +1132,6 @@ fn issue_token_ok_with_token_info_added() {
     })
 }
 
-#[test]
```
why did we drop this test?

mnaamani commented on runtime-modules/project-token/src/tests/mock.rs:
```diff
@@ -617,11 +628,19 @@ macro_rules! block {
+pub const DEFAULT_ISSUER_MEMBER_ID: u64 = 1;
 pub const DEFAULT_BLOAT_BOND: u128 = 0;
-pub const DEFAULT_INITIAL_ISSUANCE: u128 = 1_000_000;
+pub const DEFAULT_INITIAL_ISSUANCE: u128 = 100_000_000;
```
Do tests fail with smaller values?

mnaamani commented on runtime-modules/project-token/src/tests/mock.rs:
```diff
@@ -732,3 +752,8 @@ pub(crate) fn amm_function_values(
+
+// computes  supply * [(1 + 10%/100%)^(10/BlocksPerYear::get()) - 1]
+pub(crate) fn compute_correct_patronage_amount(supply: Balance) -> Balance {
+    Perquintill::from_parts(181215751000).mul_floor(supply)
```
from comment I assume this is dependant on AMM_CURVE_SLOPE and AMM_CURVE_INTERCEPT?
If so we shouldn't have a magic number `181215751000`, but base it on a const function computing it?

dobertRowneySr replied:
No this is not, it's just patronage computation

mnaamani commented on runtime-modules/project-token/src/tests/patronage.rs:
```diff
@@ -36,226 +29,142 @@ fn issue_token_ok_with_patronage_tally_count_zero() {
-        let result = Token::reduce_patronage_rate_to(token_id, decrement);
+        let result = ReducePatronageRateToFixture::default()
+            .with_target_rate(DEFAULT_YEARLY_PATRONAGE_RATE.into())
+            .execute_call();
```
we should also be testing the correct value is set, not just Ok result

mnaamani commented on runtime-modules/project-token/src/tests/patronage.rs:
```diff
@@ -264,271 +173,244 @@ fn decreasing_patronage_rate_fails_with_target_rate_exceeding_current_rate() {
+        let approx =
+            Token::account_info_by_token_and_member(DEFAULT_TOKEN_ID, DEFAULT_ISSUER_MEMBER_ID)
+                .transferrable::<Test>(System::block_number());
+        let target = 1000000181215750585898368181876;
```
I'm having a hard time accepting these pre-computed values.

dobertRowneySr replied:
fixed

mnaamani commented on runtime-modules/project-token/src/types.rs:
```diff
@@ -1,32 +1,48 @@
 use crate::{errors::Error, Config, RepayableBloatBondOf};
 
+// trait "aliases"
+// `BlockNumber` will be implemented as `u64` in the runtime configuration
```
The runtime implements BlockNumber as u32, not u64. Is this assumption here problematic?

dobertRowneySr replied:
No, it should not. According to the trait

mnaamani commented on runtime-modules/project-token/src/types.rs:
```diff
@@ -602,43 +616,14 @@ pub(crate) enum AmmOperation {
             provided_supply: Balance::zero(),
         }
     }
-    pub(crate) fn eval<T: Config>(
```
This seemed like a better place for the 'eval' implementation., but it was made into a standalone function in the content pallet. What is the reason behind this?

mnaamani commented on runtime-modules/project-token/src/types.rs:
```diff
@@ -789,9 +774,52 @@ pub enum MerkleSide {
+impl Add for YearlyRate {
+    type Output = Self;
+    fn add(self, rhs: Self) -> Self::Output {
+        Self(self.0.add(rhs.0))
```
should we be concerned with overflows?

mnaamani commented on runtime/src/utils.rs:
```diff
@@ -148,7 +148,7 @@ macro_rules! hours {
 macro_rules! days {
     ($a:expr) => {{
-        hours!(24) * $a
+        ((24 * 60 * 60 * 1000) / ExpectedBlockTime::get()) as u32 * $a
```
mathematically it looks the same to me, is this implementation more accurate (due to integer maths?)

dobertRowneySr replied:
this was in order to silence clippy, I will revisit

mnaamani commented:
I left some general comments in my last review. I think bedeho will still be in better position to review some of the test cases. I will run through full project-token implementation and perhaps do one more pass.

mnaamani commented:
I've ran weight generation and re-compiled benchmarks and ran benchmark tests successfully.

Commit:
Update runtime-modules/project-token/src/benchmarking.rs

Co-authored-by: Mokhtar Naamani <mokhtar.naamani@gmail.com>

Commit:
fix: rename

Commit:
fix: replace magic numbers in patronage

Commit:
fix: address changes

Commit:
fix: amm test

mnaamani committed:
Merge branch 'master' into update-to-polkadot-v0.9.39

Commit:
fix: make eval a method

Commit:
fix: clippy useless conversion

Commit:
fix: refactor test constants

Commit:
fix: benchmarks

Commit:
fix: content benchmarks

zeeshanakram3 committed:
Argus: fix typos

zeeshanakram3 committed:
handled exception when QN is unavailable

mnaamani committed:
setup.sh drop getsubstrate.io - tries to install deprecated protobuf apt package

mnaamani committed:
rustup install script - non-interactive shell fix

mnaamani committed:
setup.sh - rustup update after install

mnaamani committed:
setup.sh - work around outdated protobuf package install from getsubstrate.io

zeeshanakram3 committed:
fix: properly handle prematurely closed requests

mnaamani committed:
colossus: storage engine to cleanup failed upload

kdembler committed:
[Colossus] add support for elasticsearch logs auth

mnaamani committed:
colossus: multer disk storage remove comments

mnaamani committed:
colossus: handle early abort during upload

mnaamani committed:
colossus: prettier

mnaamani committed:
Merge pull request #4770 from mnaamani/update-setup-script

Fix setup.sh

kdembler committed:
improve flag parsing

kdembler committed:
update README, CHANGELOG, bump version

mnaamani committed:
update from node14 to node18

mnaamani committed:
Merge pull request #4776 from kdembler/colossus-elastic-auth

[Colossus] add support for elasticsearch logs auth

mnaamani committed:
missed some node version fixes in github workflows

mnaamani committed:
Merge branch 'master' into colossus-handle-canceled-uploads

mnaamani committed:
colossus: update changelog

zeeshanakram3 committed:
remove outdated filebeat config & Dockerfile

zeeshanakram3 committed:
add @opentelemetry/ dependencies & instrumentation

zeeshanakram3 committed:
added otlp.endpoint & otlp.attributes configuration options

zeeshanakram3 committed:
add docker-compose.elasticsearch.yml & start-elasticsearch-stack.sh

zeeshanakram3 committed:
update distributor-node/docker-compose.yml

zeeshanakram3 committed:
fix lint

mnaamani committed:
workflows: publish argus colossus docker images

mnaamani committed:
workflow: fix setting variable

mnaamani committed:
publish docker images: fix colossus.Dockerfile and use node18

mnaamani committed:
colossus argus docker builds

mnaamani committed:
fix build of images with buildx, add tests before publishing

mnaamani committed:
use correct context in docker/build-push action to reuse built artifacts

mnaamani committed:
devops: fix rustup install ansible playbook

mnaamani committed:
docker-compose fix docker image names for distributor-node

mnaamani committed:
fix auth when checking docker manifest

mnaamani committed:
skip query-node joystream/apps image

mnaamani committed:
joystream/apps image published only from master branch

mnaamani committed:
colossus: add option to not rotate logs on time basis

mnaamani committed:
fix joystream-node docker build workflow

mnaamani committed:
common entrypoint in docker-compose, publish latest tag from master

mnaamani committed:
Merge pull request #4786 from mnaamani/fix-docker-build-workflow-on-push

fix joystream-node docker build workflow

mnaamani committed:
Merge pull request #4777 from mnaamani/colossus-handle-canceled-uploads

[Colossus] handle aborted uploads

mnaamani committed:
Merge pull request #4785 from mnaamani/colossus-logs-roation-none

[Colossus] Add option to not rotate logs on time basis

mnaamani committed:
Merge branch 'master' into publish-docker-apps

mnaamani committed:
add missing mkdir dependency for storage-node

mnaamani committed:
allow additional time for QN to catchup after 10K channel creating test

zeeshanakram3 committed:
bump avj version number

zeeshanakram3 committed:
add uri validation for endpoint properties in schema

zeeshanakram3 committed:
fix: allow omitting optional config keys from config.yml when specifying them as env vars

mnaamani committed:
allow even more time for storageSync flow to wait for QN processing to complete

mnaamani committed:
storage sync wait for qn to sync

mnaamani committed:
mkdirp missing in yarn.lock

mnaamani committed:
Merge pull request #4758 from kdembler/argus-status-worker

[Argus] add workerId to status response

mnaamani committed:
fix mkdirp version

mnaamani committed:
Merge pull request #4759 from kdembler/argus-elastic-index

[Argus] allow specifying elastic logs index

mnaamani committed:
Merge pull request #4769 from zeeshanakram3/Origo_fix_unhandled_QN_exception

Origo: fix unhandled QN exception in Argus

zeeshanakram3 committed:
bump 'ajv' vesrion to ^8.0.0

mnaamani committed:
Merge pull request #4788 from zeeshanakram3/argus_fix_config_parsing_bug

[Argus] Fix config parsing bug

zeeshanakram3 committed:
Merge remote-tracking branch 'upstream/master' into origo_opentelemetry_instrumentation

zeeshanakram3 committed:
generate config docs

zeeshanakram3 committed:
update Argus CHANGELOG

zeeshanakram3 committed:
Merge remote-tracking branch 'upstream/master'

mnaamani committed:
place schema.graphql into dependant packages directory

mnaamani committed:
Merge branch 'master' into publish-docker-apps

mnaamani committed:
checkin schema.graphql in cli

mnaamani committed:
joystream-apps-docker build without depending on pre-building of query-node in monorepo

mnaamani committed:
display storage node logs when starting test storage infra

mnaamani committed:
fix build of storage-node-client in docker image for distributor

mnaamani committed:
Merge branch 'master' into update-to-node-18

mnaamani committed:
colossus: support multiple transactor accounts and verify correct keys are provided for operation

mnaamani committed:
colossus: bucketIds can be specified when running server

mnaamani committed:
colossus: if keypair cannot be unlocked, exit instead of just warning.

mnaamani committed:
integration-tests: fix renamed command arg for storage-node

mnaamani committed:
colossus: better handling of failure to unlock key files

mnaamani committed:
apps.Dockerfile -> query-node.Dockerfile

mnaamani committed:
query-node.Dockerfile build @joystream/js

mnaamani committed:
better docker layer caching

mnaamani committed:
missing deps in docker layer

mnaamani committed:
do not commit auto generated schema.graphql for cli,integration tests

mnaamani committed:
Merge pull request #4780 from Joystream/publish-docker-apps

Workflow: publish Argus, Colossus and QueryNode as docker images

mnaamani committed:
Merge branch 'master' into update-to-node-18

mnaamani committed:
update to latest yarn v1.22.19 - update some docs

mnaamani committed:
don't force new nodejs vesion, only through volta,and CI actions

mnaamani committed:
Merge branch 'master' into colossus-3.5.0

mnaamani committed:
docker-compose-no-bind-volumes use node18

zeeshanakram3 committed:
remove otel configuration from distributor-node package

zeeshanakram3 committed:
add generalized @joystream/opentelemetry package

zeeshanakram3 committed:
add otel instrumentation for query node

zeeshanakram3 committed:
add otel instrumentation for storage node

zeeshanakram3 committed:
WIP: a0515b9492 update Argus CHANGELOG

zeeshanakram3 committed:
run docker services wit otel tracing

zeeshanakram3 committed:
fix: loading otel env vars

zeeshanakram3 committed:
fix: query node build script

mnaamani committed:
ansible playbook install tools node v18

zeeshanakram3 committed:
address CR

mnaamani committed:
Merge pull request #4772 from zeeshanakram3/origo_fix_-ERR_HTTP_HEADERS_SENT

fix: properly handle prematurely closed requests

zeeshanakram3 committed:
otel: remove span names from diretibutor/storage-node instrumentations

zeeshanakram3 committed:
install @joystream/opentelemetry inside docker images

zeeshanakram3 committed:
Merge branch 'origo_opentelemetry_instrumentation' of https://github.com/zeeshanakram3/joystream into origo_opentelemetry_instrumentation

zeeshanakram3 committed:
print service token generation error in start-elasticsearch-stach.sh script

zeeshanakram3 committed:
address change requests

zeeshanakram3 committed:
fix: disable dns instrumentation in the default opentelemetry instrumentation

mnaamani committed:
fix no bind volumes docker image tests

mnaamani committed:
no docker login in pull_request flow

mnaamani committed:
docker-compose fix db name and running tests in apps workflow

zeeshanakram3 committed:
Run SP/DP/QN without telemetry in docker-compose setup

mnaamani committed:
Merge pull request #4779 from zeeshanakram3/origo_opentelemetry_instrumentation

Origo: [Argus] Opentelemetry instrumentation for better metrics & tracing

mnaamani committed:
fix typo in docker image name

mnaamani committed:
Merge pull request #4799 from mnaamani/fix-docker-push-image-name

Fix typo in docker image name

mnaamani committed:
another docker image name typo

kdembler committed:
[Argus] update elastic log level in example config

mnaamani committed:
Merge pull request #4801 from kdembler/patch-1

[Argus] update elastic log level in example config

mnaamani committed:
Merge branch 'master' into colossus-3.5.0

mnaamani committed:
colossus: update changelog

mnaamani committed:
Merge pull request #4800 from mnaamani/fix-docker-push-image-name

Fix docker image name typo

mnaamani committed:
colossus: allow server to run even if no transactor key provided for all buckets.
Uploading to bucket without transactor key will fail with a 503 status code.

mnaamani committed:
Merge branch 'master' into update-to-node-18

mnaamani committed:
colossus: allow startup with any worker id

zeeshanakram3 committed:
Merge remote-tracking branch 'upstream/master' into fix_sendExtrinsic_function_for_finalized_txs

zeeshanakram3 committed:
fix: handling of Finalized txs in sendExtrinsic function

zeeshanakram3 committed:
bump package versions & update CHANGELOG

mnaamani committed:
Merge pull request #4803 from zeeshanakram3/fix_sendExtrinsic_function_for_finalized_txs

Fix send extrinsic function for finalized txs

mnaamani committed:
Merge branch 'master' into colossus-3.5.0

mnaamani committed:
Dockerfile no need to force x86_64 platform

mnaamani committed:
install latest docker compose, uninstall old version in ansible playbook

mnaamani committed:
simpler query-node start.sh

mnaamani committed:
older docker version command does not support json

mnaamani committed:
handle old docker version command better

mnaamani committed:
configurable telemetry

mnaamani committed:
query-node start.sh, allow more time for processor to come up

mnaamani committed:
just sleep

mnaamani committed:
fix for github runner running out of diskspace

mnaamani committed:
setup.sh do not replace docker installtion, but install latest docker-compose

mnaamani committed:
fix remaining engine versions to not break things

mnaamani committed:
updated README to suggest workaround if dev machine doesn't support node18

mnaamani committed:
make binaries and chainspec easier to find

mnaamani committed:
exclude docs from code shasum calculation

mnaamani committed:
include joy-mainnet.json chainspec in joystream-node docker image

mnaamani committed:
minimum required node engine 14.18.0

mnaamani committed:
all package engine configs set to node >=14.18.0

mnaamani committed:
drop any remaining ref to joystream testnet

mnaamani committed:
add monorepo changelog

mnaamani committed:
colossus: code cleanup and revert allowing invalid workerId on startup

mnaamani committed:
Merge pull request #4804 from Joystream/docker-compose-tweaks

Docker compose tweaks

mnaamani committed:
Merge branch 'master' into update-to-node-18

mnaamani committed:
Merge branch 'master' into colossus-3.5.0

mnaamani committed:
Update CHANGELOG.md

Co-authored-by: Zeeshan Akram <37098720+zeeshanakram3@users.noreply.github.com>

mnaamani committed:
Update CHANGELOG.md

Co-authored-by: Zeeshan Akram <37098720+zeeshanakram3@users.noreply.github.com>

mnaamani committed:
Merge branch 'master' into readme-links-to-binaries

mnaamani committed:
Merge branch 'master' into package-chainspec-with-docker-image

mnaamani committed:
Merge pull request #4778 from mnaamani/update-to-node-18

Update from node v14 to v18.6.0

mnaamani committed:
Merge branch 'master' into colossus-3.5.0

mnaamani committed:
Merge pull request #4796 from mnaamani/colossus-3.5.0

Colossus 3.6.0

mnaamani committed:
Merge branch 'master' into readme-links-to-binaries

mnaamani committed:
drop reference to sudo in runtime readme

mnaamani committed:
Merge pull request #4806 from mnaamani/readme-links-to-binaries

Readme update links to binaries

mnaamani committed:
Merge branch 'master' into package-chainspec-with-docker-image

mnaamani committed:
build docker workflow: wait for server to be ready

mnaamani committed:
wait after all aws cloudformation stacks are deployed for server to be ready

mnaamani committed:
Merge pull request #4811 from mnaamani/package-chainspec-with-docker-image

Standalone joystrea/node docker image

mnaamani committed:
Merge branch 'master' into nara

mnaamani committed:
Merge branch 'master' into update-to-polkadot-v0.9.39

mnaamani committed:
Merge branch 'nara' into update-to-polkadot-v0.9.39

mnaamani committed:
update cargo-chef to work with latest nightly

mnaamani committed:
fix cargo-chef install no --locked arg

mnaamani committed:
use Weight:from_parts in place of from_ref_time which is deprecated

mnaamani committed:
generate weights using Weight:from_parts in template

mnaamani committed:
Merge pull request #4705 from Joystream/update-to-polkadot-v0.9.39

Update to polkadot v0.9.39

Commit:
fix: :rotating_light: slice indexing warning by clippy

removed the warning to allow array construction and I used iteration utilities for array transversing

Commit:
Merge branch 'nara' into crt_release

Commit:
chore: :rotating_light: update chain metadata and types

Commit:
fix: :rotating_light: post rebase fixes

Commit:
fix: :rotating_light: silence yarn cargo-checks

mnaamani commented:
@ignazio-bovo should we close this PR, it was merged into nara if I'm not mistaken?
